### PR TITLE
Add reserved keyword linter to CI

### PR DESCRIPTION
**Pull request by estute**
This adds the reserved keyword linter into the CI flow for this app. Once edx/edx-django-release-util#35 lands, I will cut a proper release of edx-django-release-util and rerun make upgrade using the pypi hosted version of the code.

There are currently violations for MYSQL keywords in this IDA, so I am only running the linter for STITCH and SNOWFLAKE keywords. In the future, if we change the field names in question, we can run the linter for MYSQL. I could also add the current violations to the whitelist/override file. I'm open to suggestions.

---

Make sure that the following steps are done before rebasing/merging

  - [ ] Request a review if desired
  - [ ] Squash/Fixup your branch to one commit
  - Config/Dependency Changes (e.g. config files, scripts that are used for provisioning etc.)
    - [ ] Make sure you have updated [edx/configuration](https://github.com/edx/configuration) and [edx/devstack](https://github.com/edx/devstack) if necessary
    - [ ] Make sure the change builds successfully in a sandbox
  - UI Changes 
    - [ ] Consider other browsers (e.g. Firefox)
    - [ ] Check mobile view
    - [ ] Consider accessibility (e.g. Run aXe on the page)

### CODE REVIEW

**Comment by macdiesel** [comment]
While I appreciate the ticket number in the title help me help future you and I by giving this PR a more descriptive name.

File: requirements/dev.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #    make upgrade
 #
 -e git+https://github.com/jsocol/django-ratelimit.git@72edbe8949fbf6699848e5847645a1998f121d46#egg=django-ratelimit  # via -r requirements/test.txt
+-e git+https://github.com/edx/edx-django-release-util.git@8bbe6a407b84f1633fd84beca1c741b08bf2435e#egg=edx-django-release-util  # via -r requirements/test.txt
```
Comment (macdiesel) [inline requirements/dev.txt]: I thought we were going to release a new release utils package so we can avoid using the -e?

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 NODE_BIN=./node_modules/.bin
 TOX = ''
 
-.PHONY: requirements upgrade piptools production-requirements all-requirements
+.PHONY: requirements upgrade piptools production-requirements all-requirements exec-check_keywords check_keywords
```
Comment (pwnage101) [inline Makefile]: Why aren't any of the other exec-* targets in this list?

File: requirements/dev.txt (hunk 1 above)
Comment (estute) [reply requirements/dev.txt]: right, once it's been released. This was just from development

**Pull request merged**